crates/yttrium/src/transaction/send/safe_test.rs

```diff
@@ -456,17 +396,74 @@ mod tests {
             tx_hash
         );
 
+        // Some extra calls to wait for/get the actual transaction. But these
+        // aren't required since eth_getUserOperationReceipt already waits
+        // let tx_hash = FixedBytes::from_slice(
+        //     &hex::decode(tx_hash.strip_prefix("0x").unwrap()).unwrap(),
+        // );
+        // let pending_txn = provider
+        //     .watch_pending_transaction(PendingTransactionConfig::new(tx_hash))
+        //     .await?;
+        // pending_txn.await.unwrap();
+        // let transaction = provider.get_transaction_by_hash(tx_hash).await?;
+        // println!("Transaction included: {:?}", transaction);
+        // let transaction_receipt =
+        //     provider.get_transaction_receipt(tx_hash).await?;
+        // println!("Transaction receipt: {:?}", transaction_receipt);
+
         Ok(user_operation_hash)
     }
 
     #[tokio::test]
     async fn test_send_transaction() -> eyre::Result<()> {
-        let transaction = Transaction::mock();
+        let rpc_url = Config::local().endpoints.rpc.base_url;
+        let rpc_url: reqwest::Url = rpc_url.parse()?;
+        let provider = ReqwestProvider::<Ethereum>::new_http(rpc_url);
+
+        let destination = LocalSigner::random();
+        let balance = provider.get_balance(destination.address()).await?;
+        assert_eq!(balance, Uint::from(0));
+
+        let owner = LocalSigner::random();
+        let sender_address = get_account_address(
+            provider.clone(),
+            Owners { owners: vec![owner.address()], threshold: 1 },
+        )
+        .await;
 
-        let transaction_hash = send_transaction(transaction).await?;
+        provider.anvil_set_balance(sender_address, U256::from(100)).await?;
+        let transaction = vec![Execution {
+            target: destination.address(),
+            value: Uint::from(1),
+            callData: Bytes::new(),
+        }];
+
+        let transaction_hash =
+            send_transaction(transaction, owner.clone()).await?;
+
+        println!("Transaction sent: {}", transaction_hash);
+
+        let balance = provider.get_balance(destination.address()).await?;
+        assert_eq!(balance, Uint::from(1));
+
+        provider.anvil_set_balance(sender_address, U256::from(100)).await?;
+        let transaction = vec![Execution {
+            target: destination.address(),
+            value: Uint::from(1),
+            callData: Bytes::new(),
+        }];
+
+        let transaction_hash = send_transaction(transaction, owner).await?;
 
         println!("Transaction sent: {}", transaction_hash);
 
+        let balance = provider.get_balance(destination.address()).await?;
+        assert_eq!(balance, Uint::from(2));
+
         Ok(())
     }
+
+    // TODO test/fix: if invalid call data (e.g. sending balance that you don't
+    // have), the account will still be deployed but the transfer won't happen.
+    // Why can't we detect this?
```

Comment:
We should know the balance of the smart account. If the smart account has not been deployed then the balance would be `0`, if it has been deployed we can get the balance of the contract before sending the user operation I believe.